crates/paas/src/persistence.rs

```diff
@@ -0,0 +1,54 @@
+//! Persistence traits for task tracking
+
+use std::time::Instant;
+
+use crate::{
+    error::ProverServiceResult,
+    program::ProgramType,
+    task::{TaskId, TaskStatus},
+};
+
+/// Task metadata for persistence
+#[derive(Debug, Clone)]
+pub struct TaskRecord<T>
+where
+    T: Clone + Eq + std::hash::Hash + std::fmt::Debug + Send + Sync + 'static,
+{
+    pub task_id: T,
+    pub uuid: String,
+    pub status: TaskStatus,
+    pub created_at: Instant,
+    pub updated_at: Instant,
+}
+
+/// Trait for persistent task storage
+///
+/// Implementations should be database-backed for production use.
+/// This trait enables idempotent task submission and crash recovery.
+pub trait TaskStore<P: ProgramType>: Send + Sync + 'static {
```

Comment:
Just want to sanity check that this gets called only in blocking-safe contexts so we don't block the executor.

Reply:
I'm not sure I see the problem, can you please elaborate? What's exactly blocking?

for now (as a least minimal measure) I'm leaving it as a TODO to handle it soon later.

Reply:
Thanks. I'm more curious what's different between the usage of db in this PR versus how we **already** access db in other services in async context (for example ASM or CSM) and what should be our best practice here. Maybe the problem is more systematic...

I'll hop on your office hours and would be glad to discuss.

Reply:
that's also relevant in the context of aux data resolver in asm (mmr db for instance) as there certainly will be db interactions.